examples/example_advanced_gsm8k.yaml

```diff
@@ -0,0 +1,136 @@
+      a) (x, y) = (0, 33.33) gives profit: <<30 * 0 + 40 * 33.33 = 1333.2>>
+      b) (x, y) = (40, 6.67) gives profit: <<30 * 40 + 40 * 6.67 = 1466.8>>
+      c) (x, y) = (35, 10) gives profit: <<30 * 35 + 40 * 10 = 1450>>
+  5. The maximum profit occurs at (40, 6.67), but we need integer solutions.
+  6. Checking the nearest integer points:
+      (40, 6) gives profit: <<30 * 40 + 40 * 6 = 1440>>
+      (39, 7) gives profit: <<30 * 39 + 40 * 7 = 1450>>
+  Therefore, the company should produce 39 units of A and 7 units of B to maximize profit.
+  #### 1450
+
+  Remember to vary the complexity and ensure all problems are solvable with the information provided.
+
+fields:
+  - name: question
+    type: str
+    description: A math problem ranging from basic to advanced, incorporating various topics and real-world contexts. This is a problem for a student, do not include the answer. Must end with a question mark.
+    validator: A natural language math problem that ends with a question mark
+    evolution_strategies:
+      - Increase the complexity by adding more steps to the problem.
+      - Introduce diverse contexts and new mathematical concepts.
+      - Enhance the wording for better clarity and engagement.
+    evolution_rate: 0.1
+
+  - name: answer
+    type: str
+    description: Detailed step-by-step solution with explanations. EVERY arithmetic or algebraic operation must be annotated with <<calculation=result>>. Must end with '#### ' followed by the final numeric answer.
+    validator: null
+    evolution_strategies:
+      - Refine the solution to improve step-by-step clarity.
+      - Add detailed annotations for all key calculations.
+      - Simplify the explanation while maintaining accuracy and thoroughness.
+    evolution_rate: 0.1
+    store_full_reflection: true
+
+# Contextual Tags
+contextual_tags:
+  tags:
+    - name: topic
+      values:
+        - arithmetic
+        - fractions
+        - percentages
+        - geometry
+        - algebra
+        - data interpretation
+        - ratios
+        - proportions
+        - compound interest
+        - polynomials
+        - exponential growth/decay
+        - probability
+        - optimization
+
+    - name: context
+      values:
+        - shopping
+        - sports
+        - cooking
+        - travel
+        - school
+        - family
+        - outdoor activities
+        - hobbies and crafts
+        - holidays and celebrations
+        - animals and nature
+
+    - name: difficulty
+      values:
+        - value: easy
+          weight: 0.1
+        - value: medium
+          weight: 0.3
+        - hard
+        - very hard
+
+    - name: age_group
+      values:
+        - value: grade 6-7
+          weight: 0.2
+        - value: grades 8-9
+          weight: 0.3
+        - grades 10-12
+
+    - name: culture
+      values:
+        - American
+        - Mexican
+        - Canadian
+        - value: European
+          weight: 0.5
```

Comment:
Q: `0.5` means that it will occur 50% less often than default (which is 1), right?

Reply:
Yes, the weights are normalized, but default is 1 for all weights. so this would mean that european attribute should be sampled .5 of the rate of one of the default values